lib/rules/no-new-nonconstructor.js

```diff
@@ -0,0 +1,64 @@
+/**
+ * @fileoverview Rule to disallow use of the new operator with global no constructor functions
+ * @author Sosuke Suzuki
+ */
+
+"use strict";
+
+//------------------------------------------------------------------------------
+// Helpers
+//------------------------------------------------------------------------------
+
+const nonConstructorGlobalFunctionNames = ["Symbol", "BigInt"];
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+/** @type {import('../shared/types').Rule} */
+module.exports = {
+    meta: {
+        type: "problem",
+
+        docs: {
+            description: "Disallow `new` operators with global no constructor functions",
+            recommended: false,
```

Comment:
Should be recommended?

Reply:
No. We can't set to recommended until we do a major release. We can switch it during the next major release.